content/blog/2022-05-31-may-22-community-gems.md

```diff
@@ -0,0 +1,188 @@
+  working with CML and GCP, DVC data and remotes, DVC pipelines and setups, and
+  more.
+descriptionLong: >
+  A roundup of technical Q&A's from the DVC and CML communities. This month:
+  working with CML and GCP, DVC data and remotes, DVC pipelines and setups, and
+  more.
+picture: 2022-05-31/may-community-gems.png
+author: milecia_mcgregor
+commentsUrl: https://discuss.dvc.org/t/may-22-community-gems/1184
+tags:
+  - DVC Remotes
+  - Pipelines
+  - CML
+  - GCP
+  - Community
+---
+
+### [Is it possible to export a plot generated using `dvc plots diff HEAD main` with multiple files to vega-lite?](https://discord.com/channels/485586884165107732/563406153334128681/965911829538832435)
+
+Thanks for the awesome question @dominic!
+
+If you are working with multiple files like `prediction.json` and
+`main.prediction.json`, you can still use the `dvc plots diff` command to get
+both of the plots to show on a single graph. You'll need to run the following
+command:
+
+```dvc
+$ dvc plots diff HEAD main --targets prediction.json main.prediction.json --json
+```
+
+We are working on expanding `dvc plots` to allow you to compare data from two
+files. You can also include this plot in a comment with CML so that it appears
+on your pull requests in GitHub.
+
+### [What is the difference between `dvc pull` and `dvc checkout`?](https://discord.com/channels/485586884165107732/563406153334128681/966739538888241192)
+
+Great question @Derek!
+
+Here are some explanations around how `dvc pull` and `dvc checkout` work.
+They're comparable to `git pull` and `git checkout`.
+
+- `dvc pull` fetches data from your remote cache and adds it to your local cache
+- `dvc checkout` syncs data from your local cache to your workspace
+
+### [Is there a way to add all of the `outs` of a `foreach` job to the `deps` of a downstream stage?](https://discord.com/channels/485586884165107732/563406153334128681/967709548393672734)
+
+Very interesting question from @mathematiguy!
+
+One way to do this is to have all `foreach` stages write out to different paths
+within the same directory and then track the entire directory as a dependency of
+your downstream stage.
+
+Here's an example of how that might look in your `dvc.yaml` file.
+
+```yaml
+stages:
+  cleanups:
+    foreach:
+      - raw1
+      - labels1
+      - raw2
+    do:
+      cmd: echo "${item}" > "data/${item}"
+      outs:
+        - data/${item}
+
+  reduce:
+    cmd: echo file > file
+    deps:
+      - data
+    outs:
+      - file
+```
+
+### [Is there a way to version and move data from one cloud storage to another with DVC remotes?](https://discord.com/channels/485586884165107732/563406153334128681/968778284114538496)
+
+Wonderful question from @Hisham!
+
+There are a couple of ways you can do this. One approach is to use
+`dvc add --to-remote`.
+
+The other approach is to use the
+[`import-url`](https://dvc.org/doc/command-reference/import-url#example-transfer-to-remote-storage)
```

Comment:
`dvc add --to-remote` (https://dvc.org/doc/command-reference/add#example-transfer-to-remote-storage) and `dvc import-url --to-remote` (https://dvc.org/doc/command-reference/import-url#example-transfer-to-remote-storage) are very similar approaches (and you need `--to-remote` in either case). 

The difference between them is noted in the docs:

> Note that you can also do this [with ](https://dvc.org/doc/command-reference/import-url#example-transfer-to-remote-storage)[dvc import-url](https://dvc.org/doc/command-reference/import-url). This has the added benefit of keeping a connection to the data source so it can be updated later (with [dvc update](https://dvc.org/doc/command-reference/update))